src/components/Menu/MenuButton.tsx

```diff
@@ -1,17 +1,44 @@
-import type { Icon } from 'lucide-react'
+'use client'
+import type { Icon } from "lucide-react";
+import Link from "next/link";
+import { usePathname } from "next/navigation";
 
 interface MenuButtonProps {
-  icon: Icon
-  isActive?: boolean
+  icon: Icon;
+  href?: string;
 }
 
-export function MenuButton({ icon: Icon, isActive = false }: MenuButtonProps) {
+export function MenuButton({ icon: Icon, href }: MenuButtonProps) {
+  const pathname = usePathname();
+  const isActive = href && pathname ? pathname.startsWith(href) : false;
+
+  if (href) {
+    return (
+      <Link href={href}>
+        <div
+          data-active={isActive}
+          className="h-12 flex justify-center items-center border-l-2 border-transparent data-[active=true]:border-[#E0DEF2]"
+        >
+          <Icon
+            strokeWidth={1.5}
+            size={28}
+            className="text-[#8F8CA8] data-[active=true]:text-[#E0DEF2]"
+          />
+        </div>
+      </Link>
+    );
+  }
+
   return (
     <div
       data-active={isActive}
       className="h-12 flex justify-center items-center border-l-2 border-transparent data-[active=true]:border-[#E0DEF2]"
     >
-      <Icon strokeWidth={1.5} size={28} color={isActive ? "#E0DEF2" : "#8F8CA8"} />
+      <Icon
+        strokeWidth={1.5}
+        size={28}
+        className="text-[#8F8CA8] data-[active=true]:text-[#E0DEF2]"
```

Comment:
Did this work? I think you need to set a `group` class in parent `div` and use `group-data-[active=true]` here.